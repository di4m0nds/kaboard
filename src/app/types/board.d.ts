/**
 * Represents a board in the task management system.
 */
export interface BoardModel {
  /**
   * Unique identifier for the board.
   */
  boardId: string;

  /**
   * The name of the board.
   */
  name: string;

  /**
   * A brief description of the board.
   */
  description: string;

  /**
   * The owner of the board.
   * Corresponds to the `userId` property in the UserModel.
   */
  ownerId: string;

  /**
   * The timestamp when the board was created.
   */
  createdAt: Date;

  /**
   * The background image associated with the board.
   */
  backgroundImage: string | null;
}


/**
 * Defines actions that can be performed on the boards in the task management system.
 */
export interface BoardActions {
  /**
   * Array containing all the boards in the system.
   */
  boards: BoardModel[];

  /**
   * Retrieves the details of a specific board based on its unique identifier.
   * @param boardId - Unique identifier of the board.
   * @returns The board details or null if not found.
   */
  getBoard: (boardId: string) => BoardModel | null;

  /**
   * Adds a new board to the task management system.
   * @param board - The board to be added.
   */
  addBoard: (board: BoardModel) => void;

  /**
   * Edits a board to the task management system.
   * @param board - The board to be edited.
   */
  editBoard: (board: BoardModel) => void;

  /**
   * Removes a board from the task management system based on its unique identifier.
   * @param boardId - Unique identifier of the board to be removed.
   */
  removeBoard: (boardId: string) => void;
}
