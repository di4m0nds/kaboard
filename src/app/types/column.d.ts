import { BoardModel } from './board'
import { StatusCode } from './types'

/**
 * Represents a column in the task management system.
 */
export interface ColumnModel {
  /**
   * Unique identifier for the column.
   */
  columnId: string;

  /**
   * The name of the column.
   */
  name: string;

  /**
   * A brief description of the column.
   */
  // description: string;

  /**
   * The unique identifier of the board to which the column belongs.
   * Corresponds to the `boardId` property in the BoardModel.
   */
  boardId: string;

  /**
   * The owner of the column.
   * Corresponds to the `userId` property in the UserModel.
   */
  owner: string;

  /**
   * The timestamp when the column was created.
   */
  createdAt: Date;

  /**
   * The color associated with the column for visual identification.
   * May be `null` if no specific color is assigned.
   */
  color: string | null;

  /**
   * The status code indicating the current state of the column.
   */
  status: StatusCode;
}

/**
 * Represents a relationship between columns in the task management system.
 */
export interface ColumnRelationshipModel {
  /**
   * Unique identifier for the column relationship.
   */
  relationshipId: string;

  /**
   * The unique identifier of the board to which the relationship belongs.
   * Corresponds to the `boardId` property in the BoardModel.
   */
  boardId: string;

  /**
   * The timestamp when the relationship was created.
   */
  createdAt: Date;

  /**
   * The unique identifier of the previous column in the board.
   * Corresponds to the `columnId` property in the ColumnModel.
   */
  previousColumnId: string;

  /**
   * The unique identifier of the next column in the board.
   * Corresponds to the `columnId` property in the ColumnModel.
   */
  nextColumnId: string;
}

/**
 * Defines actions that can be performed on the columns in the task management system.
 */
export interface ColumnActions {
  /**
   * Array containing all the columns in the system.
   */
  columns: ColumnModel[];

  /**
   * Retrieves the details of a specific column based on its unique identifier.
   * @param columnId - Unique identifier of the column.
   * @returns The column details or null if not found.
   */
  getColumn: (columnId: string) => ColumnModel | null;

  /**
   * Set columns to the task management system.
   * @param columns - The columns to be setted.
   */
  setColumns: (callback: (prevColumns: ColumnModel[]) => ColumnModel[]) => void,

  /**
   * Adds a new column to the task management system.
   * @param column - The column to be added.
   */
  addColumn: (column: ColumnModel) => void;

  /**
   * Edits a column to the task management system.
   * @param column - The column to be edited.
   */
  editColumn: (column: ColumnModel) => void;

  /**
   * Removes a column from the task management system based on its unique identifier.
   * @param columnId - Unique identifier of the column to be removed.
   */
  removeColumn: (columnId: string) => void;

  /**
   * Retrieves the details of a specific column relationship based on its unique identifier.
   * @param relationshipId - Unique identifier of the column relationship.
   * @returns The column relationship details or null if not found.
   */
  // getColumnRelationship: (relationshipId: string) => ColumnRelationshipModel | null;

  /**
   * Adds a new column relationship to the task management system.
   * @param relationship - The column relationship to be added.
   */
  // addColumnRelationship: (relationship: ColumnRelationshipModel) => void;

  /**
   * Removes a column relationship from the task management system based on its unique identifier.
   * @param relationshipId - Unique identifier of the column relationship to be removed.
   */
  // removeColumnRelationship: (relationshipId: string) => void;
}
