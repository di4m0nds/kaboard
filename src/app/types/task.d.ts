import { StatusCode } from './types'

export type TaskProps = 'taskId' | 'title' | 'description' | 'tags' | 'createdAt' | 'plannedStartDate' | 'plannedEndDate' | 'color' | 'status' | 'currentStage' | 'ownerId' | 'boardId'

/**
 * Represents a task in the application.
 */
export interface TaskModel {
  /**
   * Unique identifier for the task.
   */
  taskId: string;

  /**
   * The title of the task.
   */
  title: string;

  /**
   * A concise description of the task.
   */
  description: string;

  /**
   * Tags related of the task.
   */
  tags: string[];

  /**
   * Comments related of the task.
   */
  comments: string[];

  /**
   * The timestamp when the task was initially created.
   */
  createdAt: Date;

  /**
   * The planned start timestamp of the task timeline.
   * May be `null` if not yet determined.
   */
  plannedStartDate: Date | null;

  /**
   * The planned end timestamp of the task timeline.
   * May be `null` if not yet determined.
   */
  plannedEndDate: Date | null;

  /**
   * The color associated with the task for visual identification.
   * May be `null` if no specific color is assigned.
   */
  color: string | null;

  /**
   * The status code indicating the current state of the task.
   */
  status: StatusCode;

  /**
   * The current stage of the task, referring to a column in the board.
   */
  currentStage: string;

  /**
   * The unique identifier of the user who owns the task.
   * Corresponds to the `userId` property in the UserModel.
   */
  ownerId: string;

  /**
   * The unique identifier of the board to which the task belongs.
   * Corresponds to the `boardId` property in the BoardModel.
   */
  boardId: string;
}

/**
 * Defines actions that can be performed on the tasks in the task management system.
 */
export interface TaskActions {
  /**
   * Array containing all the tasks in the system.
   */
  tasks: TaskModel[];

  /**
   * Retrieves the details of a specific task based on its unique identifier.
   * @param taskId - Unique identifier of the task.
   * @returns The task details or null if not found.
   */
  getTask: (taskId: string) => TaskModel | null;

  /**
   * Adds a new task to the task management system.
   * @param task - The task to be added.
   */
  addTask: (task: TaskModel) => void;

  /**
   * Set taskss to the task management system.
   * @param columns - The tasks to be setted.
   */
  setTasks: (callback: (prevTasks: TaskModel[]) => TaskModel[]) => void,

  /**
   * Edits a task to the task management system.
   * @param task - The task to be edited.
   */
  editTask: (task: TaskModel) => void;

  /**
   * Removes a task from the task management system based on its unique identifier.
   * @param taskId - Unique identifier of the task to be removed.
   */
  removeTask: (taskId: string) => void;
}
