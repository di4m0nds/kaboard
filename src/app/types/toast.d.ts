/**
 * Represents the status of a toast message.
 */
export type ToastStatus = 'success' | 'error' | 'warning' | 'info';

/**
 * Defines actions related to toast messages in the application.
 */
export interface ToastActions {
  /**
   * Indicates whether the toast message is currently visible.
   */
  isToastVisible: boolean;

  /**
   * The status of the toast message (success, error, warning, info).
   */
  status: ToastStatus;

  /**
   * The heading or main content of the toast message.
   */
  heading: string;

  /**
   * Displays a toast message with the specified status and heading.
   * @param status - The status of the toast message (success, error, warning, info).
   * @param heading - The heading or main content of the toast message.
   */
  showToast: (status: ToastStatus, heading: string) => void;

  /**
   * Removes the currently displayed toast message.
   */
  removeToast: () => void;
}
