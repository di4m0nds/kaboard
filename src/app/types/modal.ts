import { ReactNode } from 'react'

/**
 * Defines actions related to modal messages in the application.
 */
export interface ModalActions {
  /**
   * Indicates whether the modal message is currently visible.
   */
  isModalVisible: boolean;

  /**
   * Node for modal content.
   */
   children: ReactNode,

  /**
   * Displays a modal with the content
   * @param children - The children for modal content.
   */
  showModal: (children: ReactNode) => void;

  /**
   * Removes the currently displayed modal message.
   */
  removeModal: () => void;
}
