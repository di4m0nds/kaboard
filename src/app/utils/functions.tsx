import { useToastStore } from "../state/toastStore"
import { ToastStatus } from "../types/toast"

export const extractKeywords = (content: string): string[] => {
  const keyWords = []
  // const wordRegex = /[$#@v*A-Za-z0-9]+/g
  // const wordRegex = /[@#$*]\w+/g
  const wordRegex = /[@#*$/][\w.-]+(\/[\w.-]+)?|\*\*.*?\*\*/g

  let match: RegExpExecArray | null
  while ((match = wordRegex.exec(content)) !== null) {
    if (match[0].includes('#') || match[0].includes('$') || match[0].includes('@') || match[0].includes('*')) {
      keyWords.push(match[0])
    }
  }

  return keyWords
}

export const applyStylingToKeywords = (content: string, color?: string | null) => {
  extractKeywords(content).forEach(word => {
    let formattedWord: string

    if (word.includes('@')) {
      formattedWord = `
      <span class="text-red-400 font-bold" >
        <a
          title="Github: ${word}"
          class="hover:underline"
          target="_blank"
          rel="noopener noreferrer"
          href="https://github.com/${word.replaceAll('@', '')}"
        >
          ${word}
        </a>
      </span>`
    } else {
      formattedWord = `<span
          class="font-extrabold"
          style="color: ${color && word.includes('#') ? `${color}` : ''}"
          title="${word.replaceAll('*','')}"
        >${word.replaceAll('*','')}</span>`
    }

    content = content.replaceAll(word, formattedWord)
  })

  return content
}

export const generateID = (): string => {
  const randomPart = Math.random().toString(36)
  return `${randomPart}`
}

