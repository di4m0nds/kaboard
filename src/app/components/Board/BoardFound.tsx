'use client'

import { BoardModel } from '@/app/types/board'
import { PlusIcon, Settings } from '../svg'
import { ColumnContentModal } from '..'
import { applyStylingToKeywords } from '@/app/utils/functions'
import { useColumnStore } from '@/app/state/columnStore'
import { useModalStore } from '@/app/state/modalStore'
import BoardContentModal from '../Modal/BoardContentModal'

interface BoardFoundProps {
  board: BoardModel
}
const BoardFound = ({ board }: BoardFoundProps) => {
  const { showModal } = useModalStore()
  const { columns } = useColumnStore()

  const { name, description, boardId } = board

  const openBoardPopUp = () => {
    showModal((
      <BoardContentModal boardId={boardId} />
    ))
  }

  const openCreateColumnPopUp = () => {
    showModal((
      <ColumnContentModal column={null} />
    ))
  }

  return (
    <>
      <header className="w-full flex justify-start items-center relative">
        <section className="w-full text-black dark:text-white">
          <h1
            className="text-4xl font-bold group"
            dangerouslySetInnerHTML={{ __html: applyStylingToKeywords(name) }}
          ></h1>
          <p
            className="mt-2 text-pretty w-full md:w-[60%] text-xl font-normal group"
            dangerouslySetInnerHTML={{ __html: applyStylingToKeywords(description) }}
          ></p>

          <nav className={`w-full ${name !== '' || description !== '' ? 'mt-5' : 'mt-0'}`}>
            <ul className="flex gap-x-2">
              <li className="flex flex-col md:flex-row items-start md:items-end">
                <button
                  data-testid="Edit Board"
                  className={`
                    group text-black dark:text-white
                    flex gap-x-2 justify-center items-center
                    px-4 py-2 rounded-xl
                    border-2 border-solid border-lime-300/70
                    hover:scale-[95%] hover:bg-lime-300/70 hover:shadow-lg
                    bg-lime-200/60
                    dark:bg-lime-600 dark:hover:bg-lime-500 
                    active:scale-100
                    duration-200
                  `}
                  onClick={openBoardPopUp}
                >
                  <Settings className="w-4 h-4 group-hover:scale-75 duration-300" />Edit Board
                </button>
              </li>
              <li className="flex flex-col md:flex-row items-start md:items-end">
                <button
                  data-testid="New Column"
                  className={`
                    group text-black dark:text-white
                    flex gap-x-2 justify-center items-center
                    px-4 py-2 rounded-xl
                    border-2 border-solid border-lime-300/70
                    hover:scale-[95%] hover:bg-lime-300/70 hover:shadow-lg
                    bg-lime-200/60
                    dark:bg-lime-600 dark:hover:bg-lime-500 
                    active:scale-100
                    duration-200
                  `}
                  onClick={openCreateColumnPopUp}
                >
                  <PlusIcon className="w-4 h-4 group-hover:scale-75 duration-300" />New Column
                </button>
                <span className="text-sm text-gray-400 ml-2">{columns.length} columns created.</span>
              </li>
            </ul>
          </nav>
        </section>
        <p className="w-full text-right text-9xl text-gray-100 dark:text-gray-800/20 font-bold hidden md:block absolute right-2 z-[-10] select-none">
          #{boardId}
        </p>
      </header>
    </>
  )
}

export default BoardFound
