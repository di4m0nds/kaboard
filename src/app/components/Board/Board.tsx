'use client'

import { useBoardStore } from '@/app/state/boardStore'
import { BoardModel } from '@/app/types/board'
import { useEffect, useState } from 'react'
import { DEFAULT_BG_IMAGE, DEFAULT_BOARD } from '@/app/configs'
import BoardFound from './BoardFound'
import ColumnList from '../Column/ColumnList'
import { useToastStore } from '@/app/state/toastStore'
import { SkeletonBoard } from '../Skeleton'
import SkeletonColumn from '../Skeleton/SkeletonColumn'

const Board = () => {
  const { showToast, removeToast } = useToastStore()
  const { boards, getBoard, addBoard } = useBoardStore()

  const [board, setBoard] = useState<BoardModel | null>(null)
  const [isLoading, setIsLoading] = useState<boolean>(true)

  setTimeout(() => setIsLoading(false), 300)

  useEffect(() => setBoard(
    getBoard(boards[0]?.boardId ?? DEFAULT_BOARD.boardId)
  ), [boards])

  const createNewBoard = () => {
    addBoard(DEFAULT_BOARD)
    showToast('success', 'New Board created.')
    setTimeout(() => removeToast(), 2000)
  }

  return (
    <section className="w-full h-auto rounded-xl shadow-sm overflow-hidden text-white max-w-[90%] mb-10 md:scale-100 scale-95">
      <div
        className="w-full h-full relative text-black dark:text-white"
        style={{
          backgroundImage: `url(${DEFAULT_BG_IMAGE})`,
          backgroundSize: 'cover',
          backgroundPosition: 'center',
        }}
      >
        <div className="absolute inset-0 bg-white/20 dark:bg-black/30"></div>
        {isLoading && (
          <div className="relative z-10 px-5 md:px-8 py-5 md:flex flex-col justify-start dark:hidden">
            <SkeletonBoard />
          </div>
        )}
        <div className="relative z-10 px-5 md:px-8 py-5 flex flex-col justify-start">
          {board && !isLoading
            ? <BoardFound board={board} />
            : !isLoading && (
              <div className="w-full flex flex-col justify-start items-start text-[#756765]/70 dark:text-zinc-200">
                <h1
                  className="font-semibold text-lg"
                >
                  No boards have been created yet.
                </h1>
                <button
                  className="text-start underline bg-transparent border-0"
                  onClick={createNewBoard}
                >
                  Click here to create new board!
                </button>
              </div>
            )}
        </div>

        {boards.length > 0 && board && <ColumnList />}
        {/*isLoading && (     COLUMN SKELETON
          <div className="relative py-3 pl-10 overflow-x-auto bg-gray-100">
            <ul className="flex gap-x-4 justify-start">
              {isLoading && [1,2,3].map((v) => <li key={v}><SkeletonColumn className="" /></li>)}
            </ul>
          </div>
        )*/}
      </div>
    </section>
  )
}

export default Board
