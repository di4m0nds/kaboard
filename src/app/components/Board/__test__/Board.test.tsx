/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom'
import { render, screen, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import Board from '@/app/components/Board/Board'
import { BoardModel } from '@/app/types/board'

import { applyStylingToKeywords, generateID } from '@/app/utils/functions'

const mockBoard: BoardModel = {
  boardId: generateID(),
  ownerId: 'di4m0nds',
  name: '',
  description: '',
  backgroundImage: null,
  createdAt: new Date(),
}

describe('Board Component', () => {
  describe('Render', () => {
    it('Renders when no boards exist', async () => {
      render(<Board />)

      await waitFor(() => {
        expect(screen.getByText(/No boards have been created yet/i)).toBeInTheDocument()
      })

      await waitFor(() => {
        expect(screen.getByText(/Click here to create new board!/i)).toBeInTheDocument()
      })
    })

    it('Should not render the "Edit Board" button', async () => {
      render(<Board />)

      expect(screen.queryByTestId('Edit Board')).not.toBeInTheDocument()
    })

    it('Should not render the "New Column" button', async () => {
      render(<Board />)

      expect(screen.queryByTestId('New Column')).not.toBeInTheDocument()
    })
  })

  describe('Behavior', () => {
    it('Renders existing board', async () => {
      const { asFragment } = render(<Board />)

      // Simulate a button click to add a default board
      await waitFor(async () => {
        const btnCreateBoard = screen.getByText(/Click here to create new board!/i)
        await userEvent.click(btnCreateBoard)
      })

      expect(asFragment()).toMatchSnapshot(applyStylingToKeywords(mockBoard.name))
      expect(asFragment()).toMatchSnapshot(applyStylingToKeywords(mockBoard.description))
    })

    it('Renders existing board and buttons', async () => {
      render(<Board />)

      expect(await screen.findByTestId('Edit Board')).toBeInTheDocument()
      expect(await screen.findByTestId('New Column')).toBeInTheDocument()
    })
  })
})
