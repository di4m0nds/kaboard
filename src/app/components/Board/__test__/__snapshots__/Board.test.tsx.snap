// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board Component Behavior Renders existing board 1`] = `
<DocumentFragment>
  <section
    class="w-full h-auto rounded-xl shadow-sm overflow-hidden text-white max-w-[90%] mb-10 md:scale-100 scale-95"
  >
    <div
      class="w-full h-full relative text-black dark:text-white"
      style="background-image: url(); background-size: cover; background-position: center;"
    >
      <div
        class="absolute inset-0 bg-white/20 dark:bg-black/30"
      />
      <div
        class="relative z-10 px-5 md:px-8 py-5 flex flex-col justify-start"
      >
        <header
          class="w-full flex justify-start items-center relative"
        >
          <section
            class="w-full text-black dark:text-white"
          >
            <h1
              class="text-4xl font-bold group"
            />
            <p
              class="mt-2 text-pretty w-full md:w-[60%] text-xl font-normal group"
            />
            <nav
              class="w-full mt-0"
            >
              <ul
                class="flex gap-x-2"
              >
                <li
                  class="flex flex-col md:flex-row items-start md:items-end"
                >
                  <button
                    class="
                    group text-black dark:text-white
                    flex gap-x-2 justify-center items-center
                    px-4 py-2 rounded-xl
                    border-2 border-solid border-lime-300/70
                    hover:scale-[95%] hover:bg-lime-300/70 hover:shadow-lg
                    bg-lime-200/60
                    dark:bg-lime-600 dark:hover:bg-lime-500 
                    active:scale-100
                    duration-200
                  "
                    data-testid="Edit Board"
                  >
                    <svg
                      aria-hidden="true"
                      class="w-4 h-4 group-hover:scale-75 duration-300"
                      fill="none"
                      viewBox="0 0 20 20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <g
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      >
                        <path
                          d="M19 11V9a1 1 0 0 0-1-1h-.757l-.707-1.707.535-.536a1 1 0 0 0 0-1.414l-1.414-1.414a1 1 0 0 0-1.414 0l-.536.535L12 2.757V2a1 1 0 0 0-1-1H9a1 1 0 0 0-1 1v.757l-1.707.707-.536-.535a1 1 0 0 0-1.414 0L2.929 4.343a1 1 0 0 0 0 1.414l.536.536L2.757 8H2a1 1 0 0 0-1 1v2a1 1 0 0 0 1 1h.757l.707 1.707-.535.536a1 1 0 0 0 0 1.414l1.414 1.414a1 1 0 0 0 1.414 0l.536-.535L8 17.243V18a1 1 0 0 0 1 1h2a1 1 0 0 0 1-1v-.757l1.707-.708.536.536a1 1 0 0 0 1.414 0l1.414-1.414a1 1 0 0 0 0-1.414l-.535-.536.707-1.707H18a1 1 0 0 0 1-1Z"
                        />
                        <path
                          d="M10 13a3 3 0 1 0 0-6 3 3 0 0 0 0 6Z"
                        />
                      </g>
                    </svg>
                    Edit Board
                  </button>
                </li>
                <li
                  class="flex flex-col md:flex-row items-start md:items-end"
                >
                  <button
                    class="
                    group text-black dark:text-white
                    flex gap-x-2 justify-center items-center
                    px-4 py-2 rounded-xl
                    border-2 border-solid border-lime-300/70
                    hover:scale-[95%] hover:bg-lime-300/70 hover:shadow-lg
                    bg-lime-200/60
                    dark:bg-lime-600 dark:hover:bg-lime-500 
                    active:scale-100
                    duration-200
                  "
                    data-testid="New Column"
                  >
                    <svg
                      aria-hidden="true"
                      class="w-4 h-4 group-hover:scale-75 duration-300"
                      fill="none"
                      viewBox="0 0 18 18"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M9 1v16M1 9h16"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                    </svg>
                    New Column
                  </button>
                  <span
                    class="text-sm text-gray-400 ml-2"
                  >
                    0 columns created.
                  </span>
                </li>
              </ul>
            </nav>
          </section>
          <p
            class="w-full text-right text-9xl text-gray-100 dark:text-gray-800/20 font-bold hidden md:block absolute right-2 z-[-10] select-none"
          >
            #0.asdfsa3
          </p>
        </header>
      </div>
      <div
        class="relative p-2 overflow-x-auto bg-gray-100/40 dark:bg-transparent"
      >
        <ul
          class="flex  gap-x-4 justify-start min-h-[100px]"
        />
      </div>
      <div
        id="DndDescribedBy-0"
        style="display: none;"
      >
        
    To pick up a draggable item, press the space bar.
    While dragging, use the arrow keys to move the item.
    Press space again to drop the item in its new position, or press escape to cancel.
  
      </div>
      <div
        aria-atomic="true"
        aria-live="assertive"
        id="DndLiveRegion-0"
        role="status"
        style="position: fixed; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%); white-space: nowrap;"
      />
    </div>
  </section>
</DocumentFragment>
`;

exports[`Board Component Behavior Renders existing board 2`] = `
<DocumentFragment>
  <section
    class="w-full h-auto rounded-xl shadow-sm overflow-hidden text-white max-w-[90%] mb-10 md:scale-100 scale-95"
  >
    <div
      class="w-full h-full relative text-black dark:text-white"
      style="background-image: url(); background-size: cover; background-position: center;"
    >
      <div
        class="absolute inset-0 bg-white/20 dark:bg-black/30"
      />
      <div
        class="relative z-10 px-5 md:px-8 py-5 flex flex-col justify-start"
      >
        <header
          class="w-full flex justify-start items-center relative"
        >
          <section
            class="w-full text-black dark:text-white"
          >
            <h1
              class="text-4xl font-bold group"
            />
            <p
              class="mt-2 text-pretty w-full md:w-[60%] text-xl font-normal group"
            />
            <nav
              class="w-full mt-0"
            >
              <ul
                class="flex gap-x-2"
              >
                <li
                  class="flex flex-col md:flex-row items-start md:items-end"
                >
                  <button
                    class="
                    group text-black dark:text-white
                    flex gap-x-2 justify-center items-center
                    px-4 py-2 rounded-xl
                    border-2 border-solid border-lime-300/70
                    hover:scale-[95%] hover:bg-lime-300/70 hover:shadow-lg
                    bg-lime-200/60
                    dark:bg-lime-600 dark:hover:bg-lime-500 
                    active:scale-100
                    duration-200
                  "
                    data-testid="Edit Board"
                  >
                    <svg
                      aria-hidden="true"
                      class="w-4 h-4 group-hover:scale-75 duration-300"
                      fill="none"
                      viewBox="0 0 20 20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <g
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      >
                        <path
                          d="M19 11V9a1 1 0 0 0-1-1h-.757l-.707-1.707.535-.536a1 1 0 0 0 0-1.414l-1.414-1.414a1 1 0 0 0-1.414 0l-.536.535L12 2.757V2a1 1 0 0 0-1-1H9a1 1 0 0 0-1 1v.757l-1.707.707-.536-.535a1 1 0 0 0-1.414 0L2.929 4.343a1 1 0 0 0 0 1.414l.536.536L2.757 8H2a1 1 0 0 0-1 1v2a1 1 0 0 0 1 1h.757l.707 1.707-.535.536a1 1 0 0 0 0 1.414l1.414 1.414a1 1 0 0 0 1.414 0l.536-.535L8 17.243V18a1 1 0 0 0 1 1h2a1 1 0 0 0 1-1v-.757l1.707-.708.536.536a1 1 0 0 0 1.414 0l1.414-1.414a1 1 0 0 0 0-1.414l-.535-.536.707-1.707H18a1 1 0 0 0 1-1Z"
                        />
                        <path
                          d="M10 13a3 3 0 1 0 0-6 3 3 0 0 0 0 6Z"
                        />
                      </g>
                    </svg>
                    Edit Board
                  </button>
                </li>
                <li
                  class="flex flex-col md:flex-row items-start md:items-end"
                >
                  <button
                    class="
                    group text-black dark:text-white
                    flex gap-x-2 justify-center items-center
                    px-4 py-2 rounded-xl
                    border-2 border-solid border-lime-300/70
                    hover:scale-[95%] hover:bg-lime-300/70 hover:shadow-lg
                    bg-lime-200/60
                    dark:bg-lime-600 dark:hover:bg-lime-500 
                    active:scale-100
                    duration-200
                  "
                    data-testid="New Column"
                  >
                    <svg
                      aria-hidden="true"
                      class="w-4 h-4 group-hover:scale-75 duration-300"
                      fill="none"
                      viewBox="0 0 18 18"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M9 1v16M1 9h16"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                    </svg>
                    New Column
                  </button>
                  <span
                    class="text-sm text-gray-400 ml-2"
                  >
                    0 columns created.
                  </span>
                </li>
              </ul>
            </nav>
          </section>
          <p
            class="w-full text-right text-9xl text-gray-100 dark:text-gray-800/20 font-bold hidden md:block absolute right-2 z-[-10] select-none"
          >
            #0.asdfsa3
          </p>
        </header>
      </div>
      <div
        class="relative p-2 overflow-x-auto bg-gray-100/40 dark:bg-transparent"
      >
        <ul
          class="flex  gap-x-4 justify-start min-h-[100px]"
        />
      </div>
      <div
        id="DndDescribedBy-0"
        style="display: none;"
      >
        
    To pick up a draggable item, press the space bar.
    While dragging, use the arrow keys to move the item.
    Press space again to drop the item in its new position, or press escape to cancel.
  
      </div>
      <div
        aria-atomic="true"
        aria-live="assertive"
        id="DndLiveRegion-0"
        role="status"
        style="position: fixed; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%); white-space: nowrap;"
      />
    </div>
  </section>
</DocumentFragment>
`;
